### PR TITLE
Added-generic-activity-filtering

### PR DESCRIPTION
**Pull request by luethiandrin**
Allows to filter the activities generically using key and value of the message header.

### CODE REVIEW

**Comment by niklas-pernter** [comment]
lgtm!

**Comment by bbortt** [comment]
@tschlat this PR interferes with your backend implementation, I think. will you add it to postfinance:feature/simulator-ui-generic-filtering as well, please? we can create one upstream pull-request after.

File: simulator-starter/src/main/java/com/consol/citrus/simulator/controller/ActivityController.java [hunk 1]
```diff
@@ -51,6 +51,11 @@ public void clearExecutions() {
         return activityService.getScenarioExecutionsByName(name);
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "/scenario/{key}={value}")
```
[the remark this answers is not in the document]
Comment (luethiandrin) [reply simulator-starter/src/main/java/com/consol/citrus/simulator/controller/ActivityController.java:54]: Why more than one header? @bbortt

File: simulator-ui/src/main/resources/static/src/app/components/activity/activity.html [hunk 2]
```diff
@@ -5,12 +5,14 @@ <h1 class="page-header"><i class="fa fa-cogs"></i> Activity
...
+    <div class="input-group-btn form-control filterInputSection">
+        <input [(ngModel)]="inputNameValue" id="name" (ngModelChange)="changeInputName()" type="text" class="nameInputField inputField" placeholder="Filter By Name">
+        <input [(ngModel)]="inputKey" id="key" (ngModelChange)="changeHeaderFields()" type="text" class="headerInputField inputField" placeholder="Head(Key)">
+        <input [(ngModel)]="inputValue" id="value" (ngModelChange)="changeHeaderFields()" type="text" class="headerInputField inputField"  placeholder="Head(Value)">
+        <button type="button" class="btn btn-primary btn-filter {{successState}} filterButton" (click)="toggleSuccess()"><i [class]="successState ? 'fa fa-check' : 'fa fa-eye-slash'"></i> Success</button>
+        <button type="button" class="btn btn-primary btn-filter {{failedState}} filterButton" (click)="toggleFailed()"><i [class]="failedState ? 'fa fa-check' : 'fa fa-eye-slash'"></i> Failed</button>
+        <button type="button" class="btn btn-primary btn-filter {{activeState}} filterButton" (click)="toggleActive()"><i [class]="activeState ? 'fa fa-check' : 'fa fa-eye-slash'"></i> Active</button>
```
[the remark this answers is not in the document]
Comment (luethiandrin) [reply simulator-ui/src/main/resources/static/src/app/components/activity/activity.html:15]: The toggle was related to the name field but it's not in use anymore.

File: simulator-ui/src/main/resources/static/src/app/components/activity/activity.ts [hunk 3]
```diff
@@ -58,11 +61,19 @@ export class ActivityComponent implements OnInit, OnDestroy, AfterViewInit {
     }
 
     getActivity() {
-        this.activityService.getScenarioExecutions(this.page, this.pageSize)
-            .subscribe(
-                scenarioExecutions => this.scenarioExecutions = scenarioExecutions,
-                error => this.errorMessage = <any>error
-            );
+        if (this.inputNameValue != "") {
```
[the remark this answers is not in the document]
Comment (luethiandrin) [reply simulator-ui/src/main/resources/static/src/app/components/activity/activity.ts:64]: Not required anymore.

**Comment by bbortt** [comment]
@luethiandrin you will tell me, once this is ready for review again?

**Comment by luethiandrin** [comment]
> @luethiandrin you will tell me, once this is ready for review again?

Yes I will. Currently waiting for Thorstens.

**Pull request closed without merging**